src/primefield/mod.rs

```diff
@@ -88,40 +91,55 @@ macro_rules! impl_from_factory {
 
 impl_from_factory!(Fp, <u8 u16 u32 u64 i8 i16 i32 i64>);
 
-macro_rules! impl_into_factory {
-    ($target:ident, <$($other:ty)+> ) => {
-     $(
-         impl IntoFactory<$target> for $other{
-            fn lift(&self, fab: $target) ->  <$target as Field>::Elt {
-                fab.elt(BigInt::from(*self))
-            }
+impl FromFactory<&str> for Fp {
+    type Output = <Fp as Field>::Elt;
+    fn from(&self, s: &str) -> Self::Output {
+        let mut sl = &s[0..];
+        if sl.len() == 0 {
+            return self.zero();
         }
-    )+
-    };
-}
-
-impl_into_factory!(Fp, <u8 u16 u32 u64 i8 i16 i32 i64>);
-
-impl<'a> FromFactory<&'a str> for Fp {
-    fn from(&self, s: &'a str) -> <Self as Field>::Elt {
-        use std::str::FromStr;
-        self.elt(BigInt::from_str(s).unwrap())
+        let mut neg = 1;
+        if &sl[0..1] == "-" {
```

Comment:
Why is it necessary to call the first index here using `[0..1]` rather than just `[0]`?

Reply:
I replaced this with the `starts_with` method.